beps/0012-metrics-service/README.md

```diff
@@ -0,0 +1,248 @@
+
+  // Future - add additional convenience methods as we learn more about the needs of the framework
+}
+```
+
+#### Root Metrics Service
+
+The `RootMetricsService` is responsible for providing metrics to other root services and creating both plugin-scoped and framework-scoped `MetricsService` instances.
+
+```ts
+interface RootMetricsService {
+  // note: no config is provided to the root service.
+  static forRoot(): RootMetricsService;
+  forPlugin(pluginId: string): MetricsService;
+  forService(serviceName: string, scope: 'plugin' | 'service'): MetricsService;
+}
+
+export const rootMetricsServiceFactory = createServiceFactory({
+  // depends on as little as possible so that it can be initialized as early as possible.
+  service: rootMetricsServiceRef,
+  deps: {},
+  factory: () => {
+    return DefaultRootMetricsService.forRoot();
+  },
+});
+```
+
+#### Plugin Metrics Service
+
+Each plugin receives a metrics service that automatically namespaces all metrics to match the naming conventions.
+
+```ts
+const metricsServiceFactory = createServiceFactory({
+  service: metricsServiceRef,
+  deps: {
+    rootMetrics: coreServices.rootMetrics,
+    pluginMetadata: coreServices.pluginMetadata,
+  },
+  factory: ({ rootMetrics, pluginMetadata }) => {
+    return rootMetrics.forPlugin(pluginMetadata.getId());
+  },
+});
+```
+
+#### Example
+
+```ts
+const entitiesProcessed = metricsService.createCounter(
+  'entities.processed.total',
+  {
+    description: 'Total entities processed during refresh',
+    unit: '{entity}',
+  },
+);
+
+entitiesProcessed.add(100);
+
+// ...
+// metric is now available as `backstage.plugin.catalog.entities.processed.total`
+```
+
+## Release Plan
+
+1. Create the new metrics-related services.
+1. Create alpha-related documentation to add to existing core service [docs](https://backstage.io/docs/backend-system/core-services/index).
+1. Release the metrics service under `@alpha`.
+1. Mark all existing metrics implementations as deprecated.
+1. Refactor catalog and scaffolder plugins to use the new (alpha) `MetricsService`.
+1. Offer a migration path for existing adopters to migrate to the new metrics service.
+1. Release the metrics service under `@public`
+1. Update remaining documentation to reference the new metrics service.
+1. Create follow-up action items to integrate the new metrics service into the core system.
+1. Fully deprecate all existing metrics implementations like the existing Prometheus one-off implementations.
+
+## Deprecation Plan
+
+1. Deprecation warning are added to all existing metrics
+2. New metrics will run in parallel with the deprecated ones for a period of time
+3. All existing metrics are removed from the codebase in the next major version
+
+## Dependencies
+
+1. The otel sdk MUST BE initialized as EARLY as possible to prevent dependents from receiving no-op meters - we will not change the current guidance on this
```

Comment:
```suggestion
1. The `otel` SDK MUST BE initialized as EARLY as possible to prevent dependents from receiving no-op meters - we will not change the current guidance on this
```